File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.go
func findMin(nums []int) int {
    l := 0
    r := len(nums) - 1

    if nums[l] < nums[r] {
        return nums[l]
    }

    for {
        m := l + ((r - l) / 2)

        if l == r {
            return nums[l]
        }

        if nums[m] > nums[r] {
            l = m + 1
        } else {
            r = m
        }
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
from sortedcontainers import SortedList

class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        w = SortedList(nums[0:k])

        print(w)

        res = [ w[k - 1] ]

        for l in range(k, len(nums)):
            w.discard(nums[l - k])

            w.add(nums[l])

            res += [ w[k - 1] ]

        return res